tests/acceptance/bootstrap/TUSContext.php

```diff
@@ -332,6 +332,43 @@ public function uploadFileUsingTus(
 		}
 	}
 
+	/**
+	 * @param string $source
+	 * @param string $destination
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function publicUploadFileUsingTus(
+		string  $source,
+		string  $destination,
+		string $password,
+	):void {
+		$password = $this->featureContext->getActualPassword($password);
+		if ($this->featureContext->isUsingSharingNG()) {
+			$token = $this->featureContext->shareNgGetLastCreatedLinkShareToken();
+		} else {
+			$token = $this->featureContext->getLastCreatedPublicShareToken();
+		}
```

Comment:
```suggestion
		$token = $this->featureContext->isUsingSharingNG() ? $this->featureContext->shareNgGetLastCreatedLinkShareToken() : $this->featureContext->getLastCreatedPublicShareToken()
```

Reply:
I think the current structure is suitable for this case, as it’s easier to read and debug. I’ll consider this approch for simple cases.